### ISSUE TITLE
Voice/Video Conference Wishlist

### ISSUE THREAD

**Issue by uhoreg** [vector-im/riot-web#3025]
Here are a bunch of things that I would like to see support for in conferencing, in order to support different uses.

- [ ] ability to join a conference as a passive participant (not sending any voice or video -- is this already possible?)
- [ ] require a certain power level to speak (e.g. for online presentations/lectures, where generally only the presenter speaks, but can also allow others to speak when needed, such as for asking questions)
- [ ] require a certain power level to send in video
- [ ] have an indicator for who is speaking (e.g. having a "foo is speaking" line in the status bar, similar to the typing indicator)
- [ ] allow moderators to mute people (telling a user's client to mute itself -- this is mainly intended as a convenience function (e.g. person has stepped away from their device and a baby starts crying loudly) and can be overridden by the user)
- [ ] persistent conferences (conferences that don't disappear when the last person leaves)
- [ ] require a certain power level to start a conference
- [ ] ability to upload a pdf/odp/ppt/etc. and use it as your "video" e.g. for presenting slides in online presentations (the same effect could be obtained by screen-sharing a viewer program, but by uploading a file and signalling to clients which slide to view, you would get both better quality and lower bandwidth usage)

**Comment by richvdh** [comment]
See also #1869

**Comment by MightyCreak** [comment]
I would add screen-sharing. This is mainly why we are still using Skype at work (there might be other solutions, but that is not the question). Surprisingly, Slack doesn't have that yet either. Screen-sharing is extremely useful: for peer-reviewing, for explaining something that is easier to explain when seen, etc...

**Comment by uhoreg** [comment]
@MightyCreak screen-sharing already exists, although it is hidden (Shift-click on the video call button), and it requires special things to be done for the browser to allow it (setting a config option in Firefox (#1312) and either a command-line argument or a custom extension (#238) for Chrome).  However, it should just work in the desktop version.

**Comment by gerg5c42g542g2c54g52c** [comment]
I'll add to this my wishlist:
I'd like to be able to create a conference (messages and voice) that I could send to people, that wouldn't require creating an account. Or in other words, the same thing that Discord has, but free software and perhaps also E2E.

**Comment by ara4n** [comment]
These are all excellent wishes.  Right now we're focused on making it work at all(!), and have just kicked off work on a new conferencing server which can hopefully evolve in this direction.  It's based on our old pre-Matrix proprietary codebase, but we're assuming we'll FOSS it if it works out okay. @superdump has the conn here :)

**Comment by PureTryOut** [comment]
It would also be nice to be able to control volume per person (user setting). From what I heard right now all streams are mixed into 1 by the homeserver, so I guess that new conferencing server could help in this.

And of course simple things like push to talk and send-on-voice-detection (although I'm not sure if the latter is possible within a web-app).

**Comment by schnuffle** [comment]
One thing that IMHO is quite important is adaptive bandwidth management. What I mean with that is that the Voice/Video part should be able to adapt to bandwidth restrictions ( static or dynamic ) by scaling the codec bandwidth used to encode the data.

**Comment by thekyriarchy** [comment]
I'd really like to see https://github.com/jitsi/jitsi-meet/issues/688 and https://github.com/jitsi/jitsi-meet/issues/30 but AIUI they would have to be handled upstream

Should this and #1869 be merged into #1965?

**Comment by uhoreg** [comment]
IMHO, this, #1869, and #1965 have slightly different scopes.  #1965 seems to be more about 1:1 calls, #1869 seems to be mostly about improving quality, while this issue is more of a long-term wishlist.

**Comment by thekyriarchy** [comment]
@uhoreg  ah, that makes sense. has the new Jitsi integration taken care of any of the items listed in these three issues?

**Comment by uhoreg** [comment]
I haven't tried the Jitsi integration, so I don't know, but looking over this list, it doesn't look like much has been fixed, aside from "ability to invite others by sending them a link without requiring a Matrix account (similar to meet.jit.si or appear.in)" (though I don't know how discoverable the link is, so it might be better to leave that open), and maybe "adaptive bandwidth management", if Jitsi does that.  @superdump would probably know better, especially regarding whether any of the issues from #1869 are fixed.

**Comment by BloodyIron** [comment]
So how soon before we can get push to talk or other stuff hammered out? Last update was Sept of last year (2017).

**Comment by BloodyIron** [comment]
Not sure how I unassigned @superdump ?_?

**Comment by ara4n** [comment]
i think that's a bug in github. @superdump is sadly no longer part of the matrix core team, and GH presumably only just noticed and unassigned. right now we're not working on new features but prioritising getting the current ones polished and stable to a 1.0 release.

**Comment by BloodyIron** [comment]
@ara4n is there a target timeline for 1.0 release? PTT is pretty paramount for a lot of riot usage that I and others are looking to do.

**Comment by thekyriarchy** [comment]
This looks interesting (video transport architecture that outperforms WebRTC, Skype, FaceTime, Hangouts, etc), albeit probably something that would need to be done upstream: https://snr.stanford.edu/salsify/

I filed an issue against Jitsi-Meet to look into it: https://github.com/jitsi/jitsi-meet/issues/2819

**Comment by MurzNN** [comment]
Screen Sharing feature are implemented more that year ago, but why it still hidden via Shift+click hack? 
Will be good to merge two buttons (Voice call, Video call) into one button, and create the popup menu with available modes: Audio call, Video call, Screen share.

**Comment by PureTryOut** [comment]
Well, I believe the default VoIP and video call systems are to be replaced by Jitsi, which already has screen sharing working as well. Wouldn't make sense imo to spend effort on the old system if it's going to be replaced (and already is on mobile) anyway.

**Comment by BloodyIron** [comment]
Wait, it's a hidden feature? WUT? Is the current screen sharing Jitsi or Freeswitch, or what?

**Comment by MurzNN** [comment]
Jitsi widget is too hard to find for regular users, so all use "Video call" button near to file upload. And using it there are no way to switch between camera & screen sharing, we can only initiate screen sharing call via shift+click, but nowhere is written about this in Riot interface.
So hope that in near future this button in Riot will be binded to Jitsi widget.

**Comment by syhe** [comment]
Does moving to Jitsi mean that for private, isolated (not federated, not relying on any external infrastructure) homeservers you need to setup an integration server (or possibly use vector.im) to get screen sharing to work?

**Comment by akontsevich** [comment]
> but why it still hidden via Shift+click hack?
Definitely: why?! Where to Shift-click to activate it? Does not work for me.

**Comment by weeman1337** [comment]
**Moderators**

At the moment the user starting the conference becomes the moderator.  
→ It would be great if additionally room moderators and up (e.g. admins) become conference moderators too.

**Comment by Thatoo** [comment]
As explain [in here](https://matrix.org/blog/2021/12/22/the-mega-matrix-holiday-special-2021#native-matrix-videovoip-conferencing), Native Group Voice/Video Conferencing is coming ad this is very exciting.

I would like to write some wishes I have for this native video conferencing system :
- allow whiteboard for participant to collaborate such as https://wbo.ophir.dev/?lang=en 
- and post-ip board such as http://scrumblr.ca/
- upload a file (pdf...) that would be a presentation support
- being able to draw/write on this presentation support
- giving the right to a participant to share his/her screen : https://github.com/vector-im/element-web/issues/16583
- allow all participant to draw/write on the shared screen (to explain something)
- allow screensharing of this whiteboard, shared screen
- a minor option would be to allow record of the conference too

**Comment by BloodyIron** [comment]
Can we PLEASE get Push To Talk?

**Comment by heyakyra** [comment]
@BloodyIron that's been requested here: https://github.com/vector-im/matrix-video-chat/issues/202

**Comment by BloodyIron** [comment]
Actually it was originally requested, by me, over 4 years ago, here : https://github.com/vector-im/element-web/issues/5993

Also, #202 was closed.

I have also brought up the request in #218 since that may be the latest issue on the matter.

**Comment by HarHarLinks** [comment]
The fact that you need a camera to even get the option to share your screen is pretty annoying and an artificial limit that doesn't make much sense.
